euphonic/force_constants.py

```diff
@@ -453,19 +453,23 @@ def _calculate_phonons_at_qpts(
         cext_err_msg = (f'Euphonic\'s C extension couldn\'t be imported '
                         f'from {euphonic_path}, it may not have been '
                         f'installed.')
-        try:
-            if use_c is not False:
-                try:
-                    import euphonic._euphonic as euphonic_c
-                except ImportError:
-                    if use_c is None:
-                        warnings.warn((
-                            cext_err_msg
-                            + ' Falling back to pure Python calculation.'),
-                            stacklevel=3)
-                    raise
-            else:
-                raise ImportError
+
+        # Check if C extension can be used and handle appropriately
+        use_c_status = False
+        if use_c is not False:
+            try:
+                import euphonic._euphonic as euphonic_c
+                use_c_status = True
+            except ImportError:
+                if use_c is None:
+                    warnings.warn((
+                        cext_err_msg
+                        + ' Falling back to pure Python calculation.'),
+                        stacklevel=3)
+                else:
+                    raise ImportCError(cext_err_msg)
+
+        if use_c_status is True:
             if n_threads is None:
                 try:
                     n_threads = int(os.environ['EUPHONIC_NUM_THREADS'])

```

Comment:
I think this will raise a ValueError if the user has set `EUPHONIC_NUM_THREADS=''`; setting to '' is a common way of pseudo-clearing an environment variable. (It will return the same thing to bash as if the variable was unset.) It will also ValueError on other invalid values (e.g. `'TWO'`) but in that case raising the error is probably best.

As well as accepting `''`, we can avoid the try/except for KeyError with `os.environ.get()`:
```python
n_threads_env = os.environ.get('EUPHONIC_NUM_THREADS', '')   # type: str
if n_threads_env:
    n_threads = int(n_threads_env)
else:
    n_threads = cpu_count()
```

Reply:
Good call, I didn't think of this case